### PR TITLE
Change ToU UX and ensure users don't access proxied resources without first agreeing to the ToU

### PR DESCRIPTION
**Pull request by uyscuti-wiki**
[T238656](https://phabricator.wikimedia.org/T238656)

### CODE REVIEW

File: TWLight/applications/templates/applications/application_evaluation.html [hunk 1]
```diff
@@ -123,6 +123,27 @@ <h2>{% trans 'Application' %}</h2>
...
+        {% endblocktrans %}
+      </strong>
+    </div>
+    <div class="col-xs-12 col-sm-8">
+      {% if user.userprofile.terms_of_use %}
+        {% trans 'Yes' %}
+      {% else %}
+        <strong><span style="color: #cc3434">{% trans 'No' %}</span></strong>
+        {% if partner_coordinator %}
+          {% comment %}Translators: This is the text displayed to coordinators when an applicant has no country of residence in their profile.{% endcomment %}
```
Comment (jsnshrmn) [inline TWLight/applications/templates/applications/application_evaluation.html]: This translation comment makes no sense to me

File: TWLight/templates/base.html [hunk 2]
```diff
@@ -215,6 +215,17 @@
         {% endblocktrans %}
       </div>
     {% endif %}
+    {% if not user.userprofile.terms_of_use and not user.is_anonymous %}
```
Comment (jsnshrmn) [inline TWLight/templates/base.html]: Does this break the "superuser passes all checks" practice?

File: TWLight/users/authorization.py [hunk 3]
```diff
@@ -481,38 +479,47 @@ def get(self, request, *args, **kwargs):
...
+                    except KeyError:
+                        return_url = reverse_lazy(
+                            "users:editor_detail", kwargs={"pk": user.editor.pk}
+                        )
+                        logger.warning(
+                            'User authenticated. No "next" parameter '
+                            "for post-login redirection."
+                        )
+                else:
+                    # Translators: This message is shown when a user logs back in to the site after their first time and hasn't agreed to the terms of use.
```
Comment (jsnshrmn) [inline TWLight/users/authorization.py]: The user should be sent on to their original destination after agreeing to terms. It looks like you're not preserving that here

File: TWLight/applications/templates/applications/application_evaluation.html (hunk 1 above)
Comment (uyscuti-wiki) [reply TWLight/applications/templates/applications/application_evaluation.html]: Forgot to update after a copy-and-paste, my bad!

File: TWLight/templates/base.html (hunk 2 above)
Comment (uyscuti-wiki) [reply TWLight/templates/base.html]: it does. Would you want me to add a superuser override?

File: TWLight/users/authorization.py (hunk 3 above)
Comment (uyscuti-wiki) [reply TWLight/users/authorization.py]: per conversation TODO for later

**Pull request merged**